Core/include/Acts/EventData/SingleCurvilinearTrackParameters.hpp

```diff
@@ -13,55 +13,121 @@
 
 namespace Acts {
 
-/// Single track parameters bound to a curvilinear reference surface.
+/// Curvilinear track parameters for a single track.
 ///
-/// @tparam charge_policy_t Selection type for the particle charge
+/// @tparam charge_t Helper type to interpret the particle charge/momentum
 ///
-/// These are just regular bound parameters with a fixed surface type.
+/// This is intended as a user-facing data class that adds additional accessors
+/// and charge/momentum interpretation on top of the pure parameters vector. All
+/// parameters and their corresponding covariance matrix are stored in
+/// curvilinear parametrization.
 ///
 /// @see SingleBoundTrackParameters
-template <typename charge_policy_t>
+template <typename charge_t>
 class SingleCurvilinearTrackParameters
-    : public SingleBoundTrackParameters<charge_policy_t> {
-  using Base = SingleBoundTrackParameters<charge_policy_t>;
+    : public SingleBoundTrackParameters<charge_t> {
+  using Base = SingleBoundTrackParameters<charge_t>;
 
  public:
   using Scalar = BoundScalar;
```

Comment:
I really wonder if we shouldn't actually use this concept generally, if I understand correctly, this comes from our Eigen definition, right?

Reply:
Let's say it is inspired by what Eigen does. The idea is to have locally defined types that encode only the semantic meaning but not e.g. the concrete data types, e.g. `Vector4 vs Vector4D`. This ensures consistency and would allow us to easily make this configurable in the future w/o having to change this everywhere.

I think it would make sense to have similar definitions at least for the event data types, but I do not think this is necessary now.

Reply:
That's something we can do in later versions, of course.